pkg/oauth/registry/oauthclientauthorization/strategy.go

```diff
@@ -79,17 +80,33 @@ func (s strategy) ValidateUpdate(ctx apirequest.Context, obj runtime.Object, old
 	oldClientAuth := old.(*oauthapi.OAuthClientAuthorization)
 	validationErrors := validation.ValidateClientAuthorizationUpdate(clientAuth, oldClientAuth)
 
-	client, err := s.clientGetter.GetClient(ctx, clientAuth.ClientName, &metav1.GetOptions{})
-	if err != nil {
-		return append(validationErrors, field.InternalError(field.NewPath("clientName"), err))
-	}
-	if err := scopeauthorizer.ValidateScopeRestrictions(client, clientAuth.Scopes...); err != nil {
-		return append(validationErrors, field.InternalError(field.NewPath("clientName"), err))
+	// only do a live client check if the scopes were increased by the update
+	if containsNewScopes(clientAuth.Scopes, oldClientAuth.Scopes) {
+		client, err := s.clientGetter.GetClient(ctx, clientAuth.ClientName, &metav1.GetOptions{})
+		if err != nil {
+			return append(validationErrors, field.InternalError(field.NewPath("clientName"), err))
+		}
+		if err := scopeauthorizer.ValidateScopeRestrictions(client, clientAuth.Scopes...); err != nil {
+			return append(validationErrors, field.InternalError(field.NewPath("clientName"), err))
+		}
 	}
 
 	return validationErrors
 }
 
+func containsNewScopes(obj []string, old []string) bool {
+	// an empty slice of scopes means all scopes, so we consider that a new scope
+	newHasAllScopes := len(obj) == 0
+	oldHasAllScopes := len(old) == 0
+	if newHasAllScopes && !oldHasAllScopes {
+		return true
+	}
+
+	newScopes := sets.NewString(obj...)
+	oldScopes := sets.NewString(old...)
+	return len(newScopes.Difference(oldScopes)) > 0
```

Comment:
(I'm not sure but) why not use `oldScopes.IsSuperset(newScopes)` instead? It won't collect the items that we don't need anyway.